app/services/charges/charge_models/volume_service.rb

```diff
@@ -50,9 +50,17 @@ def per_unit_total_amount
       end
 
       def matching_range
-        @matching_range ||= ranges.find do |range|
-          range[:from_value] <= number_of_units && (!range[:to_value] || number_of_units <= range[:to_value])
+        @matching_range ||= ranges.each_with_index do |range, index|
+          is_matching = if index.zero?
+            range[:from_value] <= number_of_units && (!range[:to_value] || number_of_units <= range[:to_value])
+          else
+            (range[:from_value] - 1) < number_of_units && (!range[:to_value] || number_of_units <= range[:to_value])
+          end
+
+          return range if is_matching
```

Comment:
It seems it could work with something like:
```suggestion
        @matching_range ||= ranges.find do |range|
          range[:from_value] <= number_of_units.ceil && (!range[:to_value] || number_of_units <= range[:to_value])
```

Reply:
Actualy yes, I over-engineered it 😄